lib/src/native/realm_core.dart

```diff
@@ -219,13 +219,29 @@ class _RealmCore {
         final syncConfigPtr = _realmLib.invokeGetPointer(() => _realmLib.realm_flx_sync_config_new(config.user.handle._pointer));
         try {
           _realmLib.realm_sync_config_set_session_stop_policy(syncConfigPtr, config.sessionStopPolicy.index);
-
+          _realmLib.realm_sync_config_set_resync_mode(syncConfigPtr, config.clientResetHandler.clientResyncMode.index);
           final errorHandlerCallback =
               Pointer.fromFunction<Void Function(Handle, Pointer<realm_sync_session_t>, realm_sync_error_t)>(_syncErrorHandlerCallback);
           final errorHandlerUserdata = _realmLib.realm_dart_userdata_async_new(config, errorHandlerCallback.cast(), scheduler.handle._pointer);
           _realmLib.realm_sync_config_set_error_handler(syncConfigPtr, _realmLib.addresses.realm_dart_sync_error_handler_callback, errorHandlerUserdata.cast(),
               _realmLib.addresses.realm_dart_userdata_async_free);
 
+          if (config.clientResetHandler is! ManualRecoveryHandler) {
```

Comment:
Yeah, I think this may have been written before establishing the type hierarchy. I think this should change to something like:

```dart
if (config.clientResetHandler.onBeforeReset != null) {
  // hook up the before reset callback in Core
}

if (config.clientResetHandler.onAfterDiscard != null || ...onAfterRecovery != null) {
  // hook up after reset callback
}
```

Reply:
Moved to https://github.com/realm/realm-dart/pull/1003